src/main/java/org/example/dao/ShortestPathDAOImpl.java

```diff
@@ -0,0 +1,82 @@
+package org.example.dao;
+
+import org.example.configurations.ConnectionPool;
+import org.example.model.Point;
+import org.example.model.ShortestPath;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class ShortestPathDAOImpl implements ShortestPathDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public ShortestPathDAOImpl() {
+    }
+
+    @Override
+    public void insertShortestPath(ShortestPath shortestPath) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try (
+             PreparedStatement statement = connection.prepareStatement("INSERT INTO shortest_paths (id, previous_point, distance) VALUES (?, ?, ?)")) {
```

Comment:
I can't view the schema right now but if there is an ID column here we should use generated keys and the while loop like in the other two DAO implementations.

Reply:
will do

Reply:
done